fhir-server/src/main/java/au/csiro/pathling/fhirpath/literal/BooleanLiteralPath.java

```diff
@@ -85,5 +84,21 @@ public boolean isComparableTo(@Nonnull final Class<? extends Comparable> type) {
   public Optional<BooleanType> getValueFromRow(@Nonnull final Row row, final int columnNumber) {
     return BooleanPath.valueFromRow(row, columnNumber);
   }
-  
+
+  @Nonnull
+  @Override
+  public BooleanLiteralPath copy(@Nonnull final String expression,
```

Comment:
This seem to be (almost) the same for all literals. Perhaps it could be refactored to LiteralPath to avoid code duplication.

Reply:
I think we can actually move `FhirPath.copy` to `NonLiteralPath`, and add validation to make it illegal to invoke a function off a literal. This would remove this override from all subclasses of `LiteralPath`.

It is not expressly prohibited by the spec, but it doesn't make a lot of sense and I can't think of a scenario where it would be needed.

Reply:
See 4a1eafea0b08b55765cd593f9c2344a613e9d579.

Reply:
See 4a1eafea0b08b55765cd593f9c2344a613e9d579.